src/main/java/uk/gov/justice/hmpps/prison/service/OffenderAddressService.java

```diff
@@ -15,13 +15,13 @@
 @Transactional(readOnly = true)
 public class OffenderAddressService {
 
-    private final OffenderBookingRepository offenderBookingRepository;
+    private final OffenderRepository offenderRepository;
     private final OffenderAddressRepository offenderAddressRepository;
 
     public List<AddressDto> getAddressesByOffenderNo(@NotNull final String offenderNo) {
-        final var offenderBooking = offenderBookingRepository.findByOffenderNomsId(offenderNo)
-            .orElseThrow(EntityNotFoundException.withMessage(String.format("No active offender found for offender number %s\n", offenderNo)));
-        final var addresses = offenderAddressRepository.findByOffenderId(offenderBooking.getOffender().getRootOffender().getId());
```

Comment:
just get the offender that was associated with last booking and get root offender from there. Guaranteed to only be one unlike the silliness above.